crates/binstalk-signal/src/lib.rs

```diff
@@ -0,0 +1,73 @@
+use std::{future::pending, io};
```

Comment:
Hmm this one feels tiny enough that I'd want to see more justification for splitting it, and maybe a rename to a generic crate, as it doesn't really have anything to do with binstall specific things.

Reply:
> Hmm this one feels tiny enough that I'd want to see more justification for splitting it

I created this one since `binstalk-downloader`'s api was depended on it.
But after some refactoring, I was able to remove the dependency in `binstalk-downloader`, so it can be put back into `binstalk`.

> and maybe a rename to a generic crate, as it doesn't really have anything to do with binstall specific things.

Something like ctrl + c signal cancellation?

I suppose this crate might be useful to other binaries.
But I can put it back if there's no use case of it outside binstall.

Reply:
Yeah, in that case let's pull it back in for now.

Reply:
> Yeah, in that case let's pull it back in for now.

Done, I've moved it back.